lib/statsd.rb

```diff
@@ -14,14 +14,77 @@
 #   statsd = Statsd.new('localhost').tap{|sd| sd.namespace = 'account'}
 #   statsd.increment 'activate'
 class Statsd
-  class RubyUdpClient
-    attr_reader :key, :sock
+  class BufferedUDPClient < UDPClient
+    @max_packet_size = 511
+    (class << self; attr_accessor :max_packet_size; end)
 
-    def initialize(address, port, key = nil)
+    def initialize(address, port)
+      super(address, port)
+      @buffer = String.new
+    end
+
+    def flush!
+      return unless @buffer.bytesize > 0
+      sock.write(@buffer)
+      @buffer.clear
+    rescue => boom
+      nil
```

Comment:
This `rescue` has been cargo-culted all the way from the original gem. I agree we should narrow this down to UDPSocket exceptions. Let me investigate if there's a definite list of those.